sqlconnect/internal/databricks/db.go

```diff
@@ -0,0 +1,107 @@
+package databricks
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+
+	databricks "github.com/databricks/databricks-sql-go"
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/sqlconnect-go/sqlconnect"
+	"github.com/rudderlabs/sqlconnect-go/sqlconnect/internal/base"
+)
+
+const (
+	DatabaseType        = "databricks"
+	defaultRudderSchema = "_rudderstack"
+	userAgentEntry      = "Rudderstack"
+)
+
+// NewDB creates a new postgres-specific client
+func NewDB(configJson json.RawMessage) (*DB, error) {
+	var config Config
+	err := config.Parse(configJson)
+	if err != nil {
+		return nil, err
+	}
+
+	connector, err := databricks.NewConnector(
+		databricks.WithAccessToken(config.Token),
+		databricks.WithServerHostname(config.Host),
+		databricks.WithPort(config.Port),
+		databricks.WithHTTPPath(config.Path),
+		databricks.WithInitialNamespace(config.Catalog, ""),
+		databricks.WithRetries(
+			config.RetryAttempts,
+			config.MinRetryWaitTime,
+			config.MaxRetryWaitTime,
+		),
+		databricks.WithUserAgentEntry(userAgentEntry),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	db := sql.OpenDB(connector)
+	db.SetConnMaxIdleTime(config.MaxConnIdleTime)
+
+	return &DB{
+		DB: base.NewDB(
+			db,
+			lo.Ternary(config.RudderSchema != "", config.RudderSchema, defaultRudderSchema),
+			base.WithDialect(dialect{}),
+			base.WithColumnTypeMapper(getColumnTypeMapper(config)),
+			base.WithJsonRowMapper(getJonRowMapper(config)),
+			base.WithSQLCommandsOverride(func(cmds base.SQLCommands) base.SQLCommands {
+				cmds.ListSchemas = func() (string, string) { return "SHOW SCHEMAS", "schema_name" }
+				cmds.SchemaExists = func(schema string) string { return fmt.Sprintf(`SHOW SCHEMAS LIKE '%s'`, schema) }
+
+				cmds.CreateTestTable = func(table string) string {
+					return fmt.Sprintf("CREATE TABLE IF NOT EXISTS %[1]s (c1 INT, c2 STRING)", table)
+				}
```

Comment:
For databricks SQL commands, I am guessing databricks defaults to lowercase which we have in dialect.go as well. Hence not using quotes?

Reply:
Table name is being passed quoted already, see tableadmin.go#15